.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,249 @@
+name: Run tests
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+  pull_request:
+
+permissions: write-all
+
+env:
+  CONTRACT_DIR: /tmp/neofs-contract
+  LOCODE_DIR: /tmp/neofs-locode-db
+
+jobs:
+  run_tests:
+    runs-on: self-hosted
+    steps:
+    # Cleanup directories before start
+    - name: Cleanup directories before start
+      run: | 
+        rm -rf ${{ env.CONTRACT_DIR }}
+        rm -rf ${{ env.LOCODE_DIR }}
+
+###############################################################################
+###############################################################################
+###############################################################################
+    # Checkout NEOFS parts for the test environment
+    # neofs-testcases
+    - name: Checkout neofs-testcases repository
+      uses: actions/checkout@v3
+      with:
+        repository: nspcc-dev/neofs-testcases
+        ref: ${{ github.event.inputs.neofs_testcases_branch || 'master' }}
+        path: neofs-testcases
+
+    # neofs-dev-env
+    - name: Checkout neofs-dev-env repository
+      uses: actions/checkout@v3
+      with:
+        repository: nspcc-dev/neofs-dev-env
+        ref: ${{ github.event.inputs.neofs_dev_env_branch || 'master' }}
+        path: neofs-dev-env
+
+    # neofs-node
+    - name: Checkout neofs-node repository
+      uses: actions/checkout@v3
+      with:
+        repository: nspcc-dev/neofs-node
+        ref: ${{ github.event.inputs.neofs_node_branch || github.head_ref || github.ref_name }}
+        path: neofs-node
+
+    # neofs-s3-gw
+    - name: Checkout neofs-s3-gw repository
+      uses: actions/checkout@v3
+      with:
+        repository: nspcc-dev/neofs-s3-gw
+        ref: ${{ github.event.inputs.neofs_s3_gw_branch || 'v0.26.1' }}
+        path: neofs-s3-gw
+
+    # neo-go
+    - name: Checkout neo-go repository
+      uses: actions/checkout@v3
```

Comment:
Do we need to checkout this and other repos like gateways? We use some released versions there that can be fetched as simple binaries.

Reply:
> Do we need to checkout this and other repos like gateways? We use some released versions there that can be fetched as simple binaries.

Neo-go is copied and used on the runner to deploy a test environment.

Reply:
So do we need a repo or a binary?